`test_synchronize_move_new_changes_to_preexisting_upstream_pr_with_multiple_commits` script test fails on MacOS very frequently

https://github.com/servo/servo/actions/runs/28617271617/job/84864136366
https://github.com/servo/servo/actions/runs/28617270985/job/84864158512
https://github.com/servo/servo/actions/runs/28605813972/job/84825648080

```
FAIL: test_synchronize_move_new_changes_to_preexisting_upstream_pr_with_multiple_commits (wpt.test.TestFullSyncRun.test_synchronize_move_new_changes_to_preexisting_upstream_pr_with_multiple_commits)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/servo/servo/python/wpt/test.py", line 556, in test_synchronize_move_new_changes_to_preexisting_upstream_pr_with_multiple_commits
    self.assertListEqual(
AssertionError: Lists differ: ['Cre[21 chars]RStep', "CommentStep:servo/servo#19612:🛠 These[115 chars]nc."] != ['Cre[21 chars]RStep:2:servo/wpt/servo_export_19612', 'OpenPR[151 chars]es.']

First differing element 0:
'CreateOrUpdateBranchForPRStep'
'CreateOrUpdateBranchForPRStep:2:servo/wpt/servo_export_19612'

Second list contains 1 additional elements.
First extra element 2:
'CommentStep:servo/servo#19612:🤖 Opened new upstream WPT pull request (wpt/wpt#1) with upstreamable changes.'

- ['CreateOrUpdateBranchForPRStep',
-  'CommentStep:servo/servo#19612:🛠 These changes could not be applied onto the '
-  "latest upstream WPT. Servo's copy of the Web Platform Tests may be out of "
-  'sync.']
+ ['CreateOrUpdateBranchForPRStep:2:servo/wpt/servo_export_19612',
+  'OpenPRStep:servo/wpt/servo_export_19612→wpt/wpt#1',
+  'CommentStep:servo/servo#19612:🤖 Opened new upstream WPT pull request '
+  '(wpt/wpt#1) with upstreamable changes.']
```

This is making it hard for PRs to land.

--- a/python/wpt/exporter/step.py
+++ b/python/wpt/exporter/step.py
@@ -111,10 +111,10 @@ class CreateOrUpdateBranchForPRStep(Step):
             # TODO: If we could cleverly parse and manipulate the full commit diff
             # we could avoid cloning the servo repository altogether and only
             # have to fetch the commit diffs from GitHub.
-            # NB: The output of git show might include binary files or non-UTF8 text,
+            # NB: The output of git might include binary files or non-UTF8 text,
             # so store the content of the diff as a `bytes`.
             diff = local_servo_repo.run_without_encoding(
-                "show", "--binary", "--format=%b", sha, "--", UPSTREAMABLE_PATH
+                "diff-tree", "--binary", "--no-commit-id", "-p", sha, "--", UPSTREAMABLE_PATH
             )
 
             # Retrieve the diff of any changes to files that are relevant

